engine/runtime/src/test/java/org/enso/interpreter/test/MetaIsATest.java

```diff
@@ -0,0 +1,118 @@
+import org.graalvm.polyglot.Value;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+
+public class MetaIsATest {
+  private Context ctx;
+  private Value isACheck;
+
+  @Before
+  public void prepareCtx() throws Exception {
+    Engine eng = Engine.newBuilder()
+      .allowExperimentalOptions(true)
+      .logHandler(new ByteArrayOutputStream())
+      .option(
+        RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+        Paths.get("../../distribution/component").toFile().getAbsolutePath()
+      ).build();
+    this.ctx = Context.newBuilder()
+      .engine(eng)
+      .allowIO(true)
+      .allowAllAccess(true)
+      .build();
+    final Map<String, Language> langs = ctx.getEngine().getLanguages();
+    assertNotNull("Enso found: " + langs, langs.get("enso"));
+
+    final URI facUri = new URI("memory://choose.enso");
+    final Source facSrc = Source.newBuilder("enso", """
+    import Standard.Base.Meta
+
+    check x y = Meta.is_a x y
+    """, "check.enso")
+            .uri(facUri)
+            .buildLiteral();
+
+    var module = ctx.eval(facSrc);
+    isACheck = module.invokeMember("eval_expression", "check");
+    assertTrue("it is a function", isACheck.canExecute());
+  }
+
+  @Test
+  public void checkNumbersAreNumber() {
+    var g = ValuesGenerator.create(ctx);
+    var typeNumber = g.typeNumber();
+    for (var v : g.numbers()) {
+      var r = isACheck.execute(v, typeNumber);
+      assertTrue("Value " + v + " is a number", r.asBoolean());
+    }
+  }
+
+  @Test
+  public void checkValuesAreAny() throws Exception {
+    var g = ValuesGenerator.create(ctx);
+    var typeAny = g.typeAny();
+    for (var v : g.allValues()) {
+      var r = isACheck.execute(v, typeAny);
+      assertTrue("Value " + v + " is any", r.asBoolean());
+    }
+  }
+
+  @Test
+  public void checkNumbersAreNotText() {
+    var g = ValuesGenerator.create(ctx);
+    for (var v : g.numbers()) {
+      var r = isACheck.execute(v, g.typeText());
+      assertFalse("Value " + v + " is not a string", r.asBoolean());
+    }
+  }
+
+  @Test
+  public void checkTextsAreText() {
+    var g = ValuesGenerator.create(ctx);
+    for (var v : g.textual()) {
+      var r = isACheck.execute(v, g.typeText());
+      assertTrue("Value " + v + " is a string", r.asBoolean());
+    }
+  }
+
+  @Test
+  public void checkTextsAreNotNumbers() {
+    var g = ValuesGenerator.create(ctx);
+    for (var v : g.textual()) {
+      var r = isACheck.execute(v, g.typeNumber());
+      assertFalse("Value " + v + " is not a number", r.asBoolean());
+    }
+  }
+
+  @Test
+  public void valuesAreNotInstancesOfThemselves() throws Exception {
+    var g = ValuesGenerator.create(ctx);
+    for (var v : g.allValues()) {
+      var r = isACheck.execute(v, v);
+      assertFalse("Value " + v + " shall not be instance of itself", r.asBoolean());
```

Comment:
I am going to change the behavior and claim that `Meta.is_a x x` shall be `False`. Of course, then https://github.com/enso-org/enso/pull/3946 has to be integrated first as otherwise the check https://github.com/enso-org/enso/pull/3946#discussion_r1040569573 would fail.